BUILDER_DEV.md

```diff
@@ -1,390 +1,294 @@
 
-## Create configuration files
-Some capabilities (such as allowing builder permissions, and turning on auto-building when new packages are uploaded to the depot) require passing in a custom config to the Builder services.
+## Officially Supported Dev Environment
 
-Create the following files somewhere on your local filesystem:
+This Dev Environment involves spinning up a virtual machine (most of the core contributors use VMWare, but Virtual Box should work as well) running Ubuntu 17.10 desktop.
 
-`config_api.toml`
-```toml
-[depot]
-path = "/hab/svc/builder-api/data"
-key_dir = "/path/to/bldr-key-pair"
+### Pre-Reqs
+
+* Ubuntu 17.10 Desktop Virtual Machine
+* Access to the "core" origin in [Production Builder](https://bldr.habitat.sh). Ask a core maintainer (or in the [Habitat Slack](http://slack.habitat.sh/)) if you do not have access.
+* Access to the Habitat 1Password account (NOTE - we are working on a solution to make this not required, but it is necessary for the time being)
+* All of the following steps should be run within your Virtual Machine
+* A [Github Personal Access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) with all repo and all user permissions.
+* The [Habitat Builder App](https://github.com/apps/habitat-builder) installed on your Github account.
+* Open your Sudoers file at /etc/sudoers and remove the "secure_path" section (this is required for the make scripts to run correctly)
+
+```
+$ sudo visudo -f /etc/sudoers
+```
+
+### Builder Setup
+* Update your system
+```
+$ sudo apt-get update
+```
+
+* Install git
+```
+$ sudo apt-get install git
+```
+
+* Install curl
+```
+$ sudo apt-get install curl
+```
+
+* Clone the [Habitat repo](https://github.com/habitat-sh/habitat)
+```
+$ git clone https://github.com/habitat-sh/habitat.git
+```
+
+* Install the [Habitat CLI](https://www.habitat.sh/docs/get-habitat/)
+* Copy habitat-builder-dev.2017-10-02.private-key.pem from 1Password into path/to/habitat/repo/.secrets/builder-github-app.pem (NOTE - this is required as of 12/11/17 - we are working on a solution to make access to the Habitat 1Password vault not required for dev environment setup).
+* Set up some environmental variables (I put these in my .bashrc)
+
+```
+export HAB_AUTH_TOKEN="<your github token>"
+export HAB_BLDR_URL="http://localhost:9636"
+export HAB_ORIGIN="core"
+```
+
+* symlink /src to path/to/your/habitat/repo
+
+```
+$ sudo ln -s path/to/your/habitat/repo /src
 ```
 
-`config_jobsrv.toml`
-```toml
-key_dir = "/path/to/bldr-key-pair"
+* Cd into your Habitat repo
 
-[archive]
-backend = "local"
-local_dir = "/tmp"
 ```
+$ cd path/to/your/habitat/repo
+```
+
+* Run the provision script
+
+```
+$ sudo ./support/linux/provision.sh
+```
+
+* After the script completes, source cargo/rust
+```
+$ source $HOME/.cargo/env
```

Comment:
It may be safer/easier to `source $HOME/.profile` since `provision.sh` may add other things as well. Plus, one would want to do this sourcing after adding the `HAB_` environment variables to the `.profile` (or `.bashrc`, which `.profile` sources) anyway.